app/ydoc-shared/src/ast/tree.ts

```diff
@@ -216,6 +204,19 @@ export abstract class Ast {
     }
   }
 
+  /** Returns child expressions and tokens. */
```

Comment:
What exactly is the definition of "child expression"? As I read the code, I see we bypass assignments for example, but not function definitions (even whose bodies are single expressions).

Reply:
This is kind of a situation-specific helper for `WidgetHierarchy`, so I moved it to there.